plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTypes.java

```diff
@@ -29,11 +36,78 @@
+import static java.lang.Math.toIntExact;
+import static java.math.RoundingMode.UNNECESSARY;
 
 public final class IcebergTypes
 {
     private IcebergTypes() {}
 
+    /**
+     * Convert value from Trino representation to Iceberg representation.
+     *
+     * @param icebergType target type to convert to
+     * @param value Trino value native for Trino type corresponding to {@code icebergType}
+     * @see TypeConverter#toTrinoType
+     */
+    public static Object convertTrinoValueToIceberg(Type icebergType, Object value)
+    {
+        if (value == null) {
+            return null;
+        }
+        if (icebergType instanceof Types.BooleanType) {
+            //noinspection RedundantCast
+            return (Boolean) value;
+        }
+        if (icebergType instanceof Types.IntegerType) {
+            return toIntExact((long) value);
+        }
+        if (icebergType instanceof Types.LongType) {
+            //noinspection RedundantCast
+            return (Long) value;
+        }
+        if (icebergType instanceof Types.FloatType) {
+            return Float.intBitsToFloat(toIntExact((long) value));
+        }
+        if (icebergType instanceof Types.DoubleType) {
+            //noinspection RedundantCast
+            return (Double) value;
+        }
+        if (icebergType instanceof Types.DecimalType icebergDecimalType) {
+            DecimalType trinoDecimalType = DecimalType.createDecimalType(icebergDecimalType.precision(), icebergDecimalType.scale());
+            BigInteger unscaledValue = trinoDecimalType.isShort()
+                    ? BigInteger.valueOf((long) value)
+                    : ((Int128) value).toBigInteger();
+            return new BigDecimal(unscaledValue, trinoDecimalType.getScale(), new MathContext(trinoDecimalType.getPrecision()));
+        }
+        if (icebergType instanceof Types.StringType) {
+            return ((Slice) value).toStringUtf8();
+        }
+        if (icebergType instanceof Types.BinaryType) {
+            return ByteBuffer.wrap(((Slice) value).getBytes());
+        }
+        if (icebergType instanceof Types.DateType) {
+            return toIntExact((long) value);
+        }
+        if (icebergType instanceof Types.TimeType) {
+            return LongMath.divide((long) value, PICOSECONDS_PER_MICROSECOND, UNNECESSARY);
+        }
+        if (icebergType instanceof Types.TimestampType) {
+            if (((Types.TimestampType) icebergType).shouldAdjustToUTC()) {
+                // timestamp with time zone
+                return Timestamps.timestampTzToMicros((LongTimestampWithTimeZone) value);
+            }
+            // timestamp
+            //noinspection RedundantCast
+            return (long) value;
+        }
+        if (icebergType instanceof Types.UUIDType) {
+            return UuidType.trinoUuidToJavaUuid((Slice) value);
+        }
+
+        throw new UnsupportedOperationException("Unsupported iceberg type: " + icebergType);
```

Comment:
Array and row types were supported before this change. Did you remove support intentionally? It would be nice to leave TODO comment if we will support them later.

Reply:
@ebyhr good catch!

- i added test coverage for ANALYZE with all supported types (extending `io.trino.plugin.iceberg.BaseIcebergConnectorTest#testAllAvailableTypes` instead of a new method, so that any new  type added is automatically test-covered)
- fixed the code to avoid failing when table has complex types; only scalar types will get analyzed
- added this to the commit message:

```
    A side effect of this change is that complex types (array, map, row) can
    no longer be analyzed: Trino can calculate a HyperLogLog for these
    types, while Iceberg does not specify binary representation for these
    types, which is required to feed data into a Theta sketch. However,
    NDV for complex types is not as useful as it is for scalar types, so
    this shouldn't matter in practice.
```

- abstained from adding TODO. i don't think it's needed.